// src/Pages/Leave/Apply.js
import React, { useState } from 'react';
import { Form, Button, Row, Col } from 'react-bootstrap';
import { employeesData } from '../AllEmployees'; 

const Apply = ({ onSubmit }) => {
    const [data, setData] = useState({
        id: '',
        name: '',
        department: '',
        startDate: '',
        endDate: '',
        typeOfLeave: '',
        reason: '',
    });

    const handleClear = () => {
        setData({
            id: '',
            name: '',
            department: '',
            startDate: '',
            endDate: '',
            typeOfLeave: '',
            reason: '',
        });
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        onSubmit(data); // Send form data to App component
        handleClear(); // Clear form after submission
    };

    const handleInputChange = (e) => {
        const { name, value } = e.target;
        setData({ ...data, [name]: value });
    };

    const handleEmployeeSelect = (e) => {
        const selectedEmployee = employeesData.find(emp => emp.id === e.target.value);
        if (selectedEmployee) {
            setData({
                ...data,
                id: selectedEmployee.id,
                name: selectedEmployee.profile,
                department: selectedEmployee.department,
            });
        }
    };

    return (
        <div className="flex items-center justify-center min-h-screen bg-gray-100">
            <div className="w-full max-w-md p-6 bg-white rounded-lg shadow-lg">
                <div className="w-full p-3 bg-blue-500 rounded-t-lg mb-4">
                    <h2 className="text-3xl font-bold text-center text-white">Leave Application Form</h2>
                    <p className="text-center text-white">Please fill in the details to apply for leave</p>
                </div>

                <Form onSubmit={handleSubmit}>
                    <Form.Group className="mb-3">
                        <Form.Label className="text-sm font-medium text-gray-900">Select Employee</Form.Label>
                        <Form.Control
                            as="select"
                            onChange={handleEmployeeSelect}
                            value={data.id}
                            required
                        >
                            <option value="" disabled>
                                -- Select Employee --
                            </option>
                            {employeesData.map(employee => (
                                <option key={employee.id} value={employee.id}>
                                    {employee.profile} ({employee.id})
                                </option>
                            ))}
                        </Form.Control>
                    </Form.Group>

                    <Form.Group className="mb-3">
                        <Form.Label className="text-sm font-medium text-gray-900">ID</Form.Label>
                        <Form.Control
                            type="text"
                            name="id"
                            value={data.id}
                            readOnly
                        />
                    </Form.Group>

                    <Form.Group className="mb-3">
                        <Form.Label className="text-sm font-medium text-gray-900">Name</Form.Label>
                        <Form.Control
                            type="text"
                            name="name"
                            value={data.name}
                            readOnly
                        />
                    </Form.Group>

                    <Form.Group className="mb-3">
                        <Form.Label className="text-sm font-medium text-gray-900">Department</Form.Label>
                        <Form.Control
                            type="text"
                            name="department"
                            value={data.department}
                            readOnly
                        />
                    </Form.Group>

                    <Form.Group className="mb-3">
                        <Form.Label className="text-sm font-medium text-gray-900">Start Date</Form.Label>
                        <Form.Control
                            type="date"
                            name="startDate"
                            value={data.startDate}
                            onChange={handleInputChange}
                            required
                        />
                    </Form.Group>

                    <Form.Group className="mb-3">
                        <Form.Label className="text-sm font-medium text-gray-900">End Date</Form.Label>
                        <Form.Control
                            type="date"
                            name="endDate"
                            value={data.endDate}
                            onChange={handleInputChange}
                            required
                        />
                    </Form.Group>

                    <Form.Group className="mb-3">
                        <Form.Label className="text-sm font-medium text-gray-900">Type of Leave</Form.Label>
                        <Form.Control
                            as="select"
                            name="typeOfLeave"
                            value={data.typeOfLeave}
                            onChange={handleInputChange}
                            required
                        >
                            <option value="" disabled>
                                -- Please Select --
                            </option>
                            <option value="Annual Leave">Annual Leave</option>
                            <option value="Casual Leave">Casual Leave</option>
                            <option value="Medical Leave">Medical Leave</option>
                            <option value="Maternity Leave">Maternity Leave</option>
                        </Form.Control>
                    </Form.Group>

                    <Form.Group className="mb-3">
                        <Form.Label className="text-sm font-medium text-gray-900">Reason for Leave</Form.Label>
                        <Form.Control
                            as="textarea"
                            name="reason"
                            value={data.reason}
                            onChange={handleInputChange}
                            placeholder="State the reason for leave here..."
                        />
                    </Form.Group>

                    <Row className="justify-content-end">
                        <Col xs="auto">
                            <Button variant="danger" className="me-2" onClick={handleClear}>
                                Clear
                            </Button>
                            <Button type="submit" variant="success">
                                Apply Leave
                            </Button>
                        </Col>
                    </Row>
                </Form>
            </div>
        </div>
    );
};

export default Apply;
